components/brave_rewards/resources/ui/components/donate/index.tsx

```diff
@@ -69,18 +69,52 @@ export default class Donate extends React.PureComponent<Props, State> {
     if (this.sendButton) {
       this.sendButton.focus()
     }
+
+    this.donationAmountsDidChange()
+  }
+
+  donationAmountsAreEqual = (a: Donation[], b: Donation[]) => {
+    if (a.length !== b.length) {
+      return false
+    }
+
+    return a.every(({}, index) => a[index].tokens === b[index].tokens)
   }
 
   componentDidUpdate (prevProps: Props) {
-    if (
-      this.props.balance !== prevProps.balance ||
-      this.props.donationAmounts !== prevProps.donationAmounts ||
-      this.props.currentAmount !== prevProps.currentAmount
-    ) {
+    if (!this.donationAmountsAreEqual(this.props.donationAmounts, prevProps.donationAmounts)) {
+      this.donationAmountsDidChange()
+      return
+    }
+
+    if (this.props.balance !== prevProps.balance ||
+        this.props.currentAmount !== prevProps.currentAmount) {
       this.validateAmount(this.props.balance)
     }
   }
 
+  donationAmountsDidChange = () => {
+    const amounts = this.props.donationAmounts
+    if (this.state.amountChanged) {
+      // If the user has selected an amount that is no longer in the donation amounts, clear the
+      // selection.
+      if (this.props.onAmountSelection &&
+          !amounts.some(amount => amount.tokens === this.props.currentAmount)) {
+        this.props.onAmountSelection('0.0')
```

Comment:
shouldn't we in this case select middle one again?

Reply:
After discussion with @NejcZdovc, I changed this to not modify the options presented in the UI one the user has made a selection. This should be rare as it should only happen when the request to get wallet properties is significantly delayed.